### DISCUSSION TITLE
How do I map a schema to an existing kotlin class when generating an openapi server?

### DISCUSSION THREAD

**Discussion by ricker-flow** [micronaut-projects/micronaut-openapi#1783] [Q&A] [answered]
How do I reference an existing kotlin class when generating an openapi server? I tried using importMapping but that does not work.

**Comment by altro3** [comment]
Maybe it's a bug, or maybe you're doing something wrong.

In general, this is done using standard openapi generator tools. Read the documentation on the website https://openapi-generator.tech/docs/generators/kotlin/ , try generating code using the standard openaApiGenerator plugin, and see what happens. If the standard generator plugin generates correct code, and micronaut-openapi generates incorrect code, then it looks more like a bug.

It's better to immediately write about your problem, suggestion, or found bug in the issue: https://github.com/micronaut-projects/micronaut-openapi/issues

I rarely look through discussions

**Comment by ricker-flow** [comment, marked as answer]
I figured it out 

```
components:
  schemas:    
    MyObject:
      type: object
      properties:
        start:
          $ref: "MySpecialClass"
```
          
```
server("mymodel", file("src/main/resources/omymodel-api.yaml")) {
  apiPackageName = "foo.bar.api"
  modelPackageName = "foo.bar.dto"
  useReactive = false
  useAuth = false
  lang = "kotlin"
  importMapping =
    mapOf(
      "MySpecialClass" to "foo.bar.MySpecialClass",
    )
}
```
